### DISCUSSION TITLE
Feature Request: Support useTempFile for express-fileupload

### DISCUSSION THREAD

**Discussion by BitPatty** [payloadcms/payload#746] [Feature Requests & Ideas]
# Bug Report

To avoid excessive memory usage for uploads `useTempFiles` for express-fileupload can be enabled in the payload config. This of course flushes the buffer on the `req.files.file` object on which payload is relying for resizing.

## Expected Behavior

`useTempFiles` should not affect payloads upload mechanic.

## Current Behavior

Payload fails to process the image since it expects the whole image to be in a buffer.

## Possible Solution

Check whether the temp file exists and do further processing based on this structure.

+ Resizes images should also not be kept as a buffer if `useTempFiles` is true. Depending on the source file and resizing options the memory usage is a real concern.

## Steps to Reproduce

1. `upload.useTempFiles` to `true` in the payload config
2. Upload an image

## Detailed Description

Payload 0.19.2

https://github.com/payloadcms/payload/blob/cac5266c79dd6b2338077eab2eda3274dd7db606/src/uploads/uploadFile.ts#L59-L87

**Comment by denolfe** [comment]
Hey @BitPatty , thanks for the report. Since we used express-fileupload's TS type for `upload`, it looks like this feature would be supported in Payload. However, this feature is not intended to be supported.

I think this would be a great enhancement, though. I'll move this to discussions as a feature request.

**Reply by BitPatty** [reply]
Is there an existing "built-in" way to override the whole upload function and/or the resizing mechanic?

Currently, the `beforeChange` hook is triggered too late to intervene and express-fileupload has no extension points either.
